ansible_bender/callback_plugins/snapshoter.py

```diff
@@ -52,26 +56,57 @@ def _snapshot(self, task_result):
             return
         if not build.is_layering_on():
             return
-        content = self.get_task_content(task_result._task)
         if task_result.is_skipped() or getattr(task_result, "_result", {}).get("skip_reason", False):
-            a.record_progress(None, content, None, build_id=build.build_id)
+            a.record_progress(None, self._task_content, None, build_id=build.build_id)
             return
         # # alternatively, we can guess it's a file action and do getattr(task, "src")
         # # most of the time ansible says changed=True even when the file is the same
         if task_result._task.action in FILE_ACTIONS:
             if not task_result.is_changed():
-                status = a.maybe_load_from_cache(content, build_id=build.build_id)
+                status = a.maybe_load_from_cache(self._task_content, build_id=build.build_id)
                 if status:
                     self._display.display("loaded from cache: '%s'" % status)
                     return
-        image_name = a.cache_task_result(content, build)
+        image_name = a.cache_task_result(self._task_content, build)
         if image_name:
             self._display.display("caching the task result in an image '%s'" % image_name)
 
     @staticmethod
     def get_task_content(task: Task):
+
+        def get_templated_ds(task: Task):
```

Comment:
could you please move this function out of the class (e.g. to the top of this file) and please specify in the docstring it's meant to be used in this method